### ISSUE TITLE
Incompatible function pointer types sending 'unsigned char'

### ISSUE THREAD

**Issue by shkumbinhasani** [itinance/react-native-sha256#70]
My IOS pipeline started failing with this error.

```

Error
Incompatible function pointer types sending 'unsigned char *(const void *, CC_LONG, unsigned char *)' (aka 'unsigned char *(const void *, unsigned int, unsigned char *)') to parameter of type 'DIGEST_FUNCTION' (aka 'unsigned char (*)(const void *, unsigned int, unsigned char *)')

sha256.m:74

Error
Incompatible function pointer types sending 'unsigned char *(const void *, CC_LONG, unsigned char *)' (aka 'unsigned char *(const void *, unsigned int, unsigned char *)') to parameter of type 'DIGEST_FUNCTION' (aka 'unsigned char (*)(const void *, unsigned int, unsigned char *)')

sha256.m:97

Error
Incompatible function pointer types sending 'unsigned char *(const void *, CC_LONG, unsigned char *)' (aka 'unsigned char *(const void *, unsigned int, unsigned char *)') to parameter of type 'DIGEST_FUNCTION' (aka 'unsigned char (*)(const void *, unsigned int, unsigned char *)')

sha256.m:88

Error
Incompatible function pointer types sending 'unsigned char *(const void *, CC_LONG, unsigned char *)' (aka 'unsigned char *(const void *, unsigned int, unsigned char *)') to parameter of type 'DIGEST_FUNCTION' (aka 'unsigned char (*)(const void *, unsigned int, unsigned char *)')

sha256.m:65
~```

**Comment by teaualune** [comment]
After upgrading my Xcode to 15.3 I get this error as well.

Remove the parentheses around `*DIGEST_FUNCTION` in `sha256.m` and it's all good to go:

```objective-c
typedef unsigned char *DIGEST_FUNCTION(const void *data, CC_LONG len, unsigned char *md);
```

**Comment by smithmf** [comment]
Thanks for the info @teaualune! worked for me after an Xcode update today.

**Comment by itinance** [comment]
#71 published in v1.4.10

**Comment by coawazie** [comment]
Seems this issue is back!
